libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2242,6 +2244,32 @@ MAV_RESULT GCS_MAVLINK::handle_command_set_message_interval(const mavlink_comman
     return MAV_RESULT_FAILED;
 }
 
+/*
+  this function is reserved for use by scripting
+ */
+MAV_RESULT GCS::set_message_interval(uint8_t port_num, uint32_t msg_id, int32_t interval_us)
+{
+    uint8_t channel = get_channel_from_port_number(port_num);
+
+    if (channel < MAVLINK_COMM_NUM_BUFFERS && chan(channel) != nullptr)
+    {
+        chan(channel)->set_message_interval(msg_id, interval_us);
```

Comment:
there should be a "return MAV_RESULT_ACCEPTED;" after this line I think or otherwise it will always return MAV_RESULT_FAILED.